driver-core/src/test/functional/com/mongodb/JsonTestServerVersionChecker.java

```diff
@@ -43,6 +43,10 @@ public static boolean skipTest(final BsonDocument testDocument, final BsonDocume
     }
 
     private static boolean canRunTest(final BsonDocument document, final ServerVersion serverVersion) {
+        if (document.containsKey("serverless") && (!serverlessMatches(document.getString("serverless").getValue()))) {
```

Comment:
Handle serverless restrictions in legacy tests.

Reply:
I went with

        if ((!serverlessMatches(document.getString("serverless", new BsonString("allow")).getValue()))) {
            return false;
        }

instead, as I prefer not to pass BsonString parameters to places that shouldn't have to care about BSON.

Reply:
Done